### PR TITLE
Bump `pylint` to 2.8.0

### PR DESCRIPTION
**Pull request by jiasli**
Replace https://github.com/Azure/azure-cli-dev-tools/pull/294
Bump pylint to 2.8.0 to support Python 3.9 (https://github.com/Azure/azure-cli/pull/17368)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -61,23 +61,22 @@
...
+        'pylint==2.8.0',
+        'pytest-xdist',  # depends on pytest-forked
         'pytest>=5.0.0',
-        'pytest-xdist', # depends on pytest-forked
         'pyyaml',
         'requests',
         'sphinx==1.6.7',
         'tox',
         'wheel==0.30.0',
-        'azure-multiapi-storage',
```
Comment (kairu-ms) [inline setup.py:79]: You cannot remove it. It's used in extension release command.

File: setup.py [hunk 2]
```diff
@@ -61,23 +61,22 @@
         'azdev.utilities',
     ],
     install_requires=[
+        'azure-multiapi-storage',
         'docutils',
         'flake8',
         'gitpython',
+        'isort==4.3.21'
```
Comment (kairu-ms) [inline setup.py]:
```suggestion
        'isort==4.3.21',
```

File: azdev/operations/tests/test_benchmark.py [hunk 3]
```diff
@@ -158,8 +158,8 @@ def test_benchmark_with_help_command(self):
 
     def test_benchmark_in_actual_running(self):
         with mock.patch(
-            "multiprocessing.pool.Pool.map_async",  # pylint: disable=bad-continuation
```
Comment (jiasli) [inline azdev/operations/tests/test_benchmark.py:161]: `bad-continuation` has been removed: http://pylint.pycqa.org/en/latest/whatsnew/2.6.html

File: azdev/operations/extensions/util.py [hunk 4]
```diff
@@ -46,9 +46,8 @@ def _get_azext_metadata(ext_dir):
 def get_ext_metadata(ext_dir, ext_file, ext_name):
     # Modification of https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/extension.py#L89
     WHL_METADATA_FILENAME = 'metadata.json'
-    zip_ref = zipfile.ZipFile(ext_file, 'r')
-    zip_ref.extractall(ext_dir)
-    zip_ref.close()
+    with zipfile.ZipFile(ext_file, 'r') as zip_ref:
+        zip_ref.extractall(ext_dir)
```
Comment (jiasli) [inline azdev/operations/extensions/util.py:50]: Fix: 

> Consider using 'with' for resource-allocating operations (consider-using-with)

File: .pylintrc [hunk 5]
```diff
@@ -1,17 +1,18 @@
...
 # For all codes, run 'pylint --list-msgs' or go to 'http://pylint-messages.wikidot.com/all-codes'
 disable=
-    invalid-name,
+    cyclic-import,
     fixme,
+    import-outside-toplevel,
+    invalid-name,
     missing-docstring,
-    too-many-arguments,
+    raise-missing-from,
```
Comment (jiasli) [inline .pylintrc:9]: This should not be forced: E0012: Bad option value 'bad-continuation' (bad-option-value)

File: azdev/operations/performance.py [hunk 6]
```diff
@@ -162,6 +162,7 @@ def benchmark(commands=None, runs=20):
     for raw_command in commands:
         logger.info("Measuring %s...", raw_command)
 
+        # pylint: disable=consider-using-with
```
Comment (jiasli) [inline azdev/operations/performance.py:165]: `azdev/operations/performance.py` is unfinished and out of maintenance.

**Review by kairu-ms — APPROVED** [review]

File: .pylintrc (hunk 5 above)
Comment (jiasli) [inline .pylintrc:9]: `raise-missing-from` should not be forced: 

> W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)

File: .pylintrc [hunk 7]
```diff
@@ -1,17 +1,18 @@
 [MESSAGES CONTROL]
 # For all codes, run 'pylint --list-msgs' or go to 'http://pylint-messages.wikidot.com/all-codes'
 disable=
-    invalid-name,
+    cyclic-import,
```
Comment (qwordy) [inline .pylintrc:4]: I recommend adding explanation of why disable them.

Comment (jiasli) [reply .pylintrc:4]: Good suggestion, but let's keep it for now as this is only for `azdev`. CLI's `pylintrc` has detailed explanation.

File: .pylintrc [hunk 8]
```diff
@@ -1,17 +1,18 @@
...
+    invalid-name,
     missing-docstring,
-    too-many-arguments,
+    raise-missing-from,
     too-few-public-methods,
-    cyclic-import,
-    useless-object-inheritance,
+    too-many-arguments,
     useless-import-alias,
+    useless-object-inheritance,
```
Comment (qwordy) [inline .pylintrc]:
```
class A(object):
    pass

class A:    # better
    pass
```

Comment (qwordy) [inline .pylintrc]:
```suggestion
```

Comment (jiasli) [reply .pylintrc]: Fixed.

Comment (jiasli) [reply .pylintrc]: Fixed.

File: .pylintrc (hunk 7 above)
Comment (jiasli) [reply .pylintrc:4]: I have removed unused items. The rest of them is self-explanatory.

File: setup.py [hunk 9]
```diff
@@ -61,23 +61,21 @@
...
+        'pytest-xdist',  # depends on pytest-forked
         'pytest>=5.0.0',
-        'pytest-xdist', # depends on pytest-forked
         'pyyaml',
         'requests',
         'sphinx==1.6.7',
         'tox',
-        'wheel==0.30.0',
-        'azure-multiapi-storage',
-        'isort==4.3.21'
```
Comment (kairu-ms) [inline setup.py:80]: Why remove 'isort==4.3.21'

**Review by qwordy — APPROVED** [review]

File: setup.py (hunk 9 above)
Comment (jiasli) [reply setup.py:80]: `isort` was pinned by #225 due to a similar issue as astroid (#294).

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**